examples/cell-range-selection/src/App.js

```diff
@@ -0,0 +1,173 @@
+import { useCellRangeSelection } from 'react-table-plugins'
+import makeData from './makeData'
+
+const Styles = styled.div`
+  padding: 1rem;
+
+  table {
+    height: 200px;
+    border-spacing: 0;
+    border: 1px solid black;
+
+    tr {
+      :last-child {
+        td {
+          border-bottom: 0;
+        }
+      }
+    }
+
+    th,
+    td {
+      margin: 0;
+      padding: 0.5rem;
+      border-bottom: 1px solid black;
+      border-right: 1px solid black;
+
+      :last-child {
+        border-right: 0;
+      }
+    }
+  }
+`
+
+function Table ({ columns, data }) {
+  // Use the state and functions returned from useTable to build your UI
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    // currentSelectedCellIds: cells of a current selected range
+    // selectedCellIds: All previously selected cells
+    state: { selectedCellIds, currentSelectedCellIds },
+    // getCellsBetweenId (Fn): Pass two cell Ids to get all cell Ids between them
+    getCellsBetweenId,
+    setSelectedCellIds,
+    cellsById
+  } = useTable(
+    {
+      columns,
+      data,
+      // cellIdSplitBy (string): Cell id is split by column.id + cellIdSplitBy + row.id
+      cellIdSplitBy: 'cols_rows',
+      initialState: {
+        selectedCellIds: {}
+      }
+    },
+    useCellRangeSelection
+  )
+
+  let cellsSelected = { ...currentSelectedCellIds, ...selectedCellIds }
+
+  // returns two random cell ids, this is just for the demo.
+  const getRandomCellIds = React.useCallback(() => {
+    let cloneCellIds = Object.keys(cellsById)
+    let randomCellId = () =>
+      cloneCellIds[(cloneCellIds.length * Math.random()) << 0]
+    return [randomCellId(), randomCellId()]
+  }, [cellsById])
+
+  // getCellsBetweenId returns all cell Ids between two cell Id, and then setState for selectedCellIds
+  const setCells = React.useCallback(() => {
+    const cellsBetween = getCellsBetweenId(...getRandomCellIds())
+    setSelectedCellIds(cellsBetween)
+  }, [getCellsBetweenId, setSelectedCellIds, getRandomCellIds])
+
+  return (
+    <>
+      <button onClick={setCells}>setState selectedCellIds</button>
+      <table {...getTableProps()}>
+        <thead>
+          {headerGroups.map(headerGroup => (
+            <tr {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map(column => (
+                <th {...column.getHeaderProps()}>{column.render('Header')}</th>
+              ))}
+            </tr>
+          ))}
+        </thead>
+        <tbody {...getTableBodyProps()}>
+          {rows.map((row, i) => {
+            prepareRow(row)
+            return (
+              <tr {...row.getRowProps()}>
+                {row.cells.map(cell => {
+                  return (
+                    <td
+                      {...cell.getCellRangeSelectionProps()}
+                      {...cell.getCellProps()}
+                      style={
+                        cellsSelected[cell.id]
```

Comment:
Can we add another property in `cell` itself something like isSelected(if it doesn't exists) and use `cell.isSelected` here

Reply:
Yeah, actually I tried it [Here](https://github.com/gargroh/react-table-plugins/pull/3/commits/fb36ae2fcc51227eef61e3dd0d82c15323e6d82a)

If I am correct, `row.allCells (all visible cells)` property is added in the `prepareRow`, So I get `undefined` in my `useInstance hook`. 

Adding this in `prepareRow` will work but, it runs only at the last and I get to see on UI after I end my selecting the range. 

I am not sure if I am adding it to the right hook (`useInstance`). So I am unable to add it. Let me know your thoughts and how I should be approaching this.

Also, As `useInstance` will trigger on each cell selects, adding `isSelected` runs loop 500 times (for eg. 100rows * 5 columns)

Reply:
>adding isSelected runs loop 500 times (for eg. 100rows * 5 columns) 
ahh..if that's the problem...then lets not go with tht approach...